File: TENavigationChildVC/TENavigationChildVC/ViewController.m
//
//  ViewController.m
//  TENavigationChildVC
//
//  Created by bo on 08/06/2017.
//  Copyright © 2017 TE. All rights reserved.
//

#import "ViewController.h"
#import "ViewController2.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"left" style:UIBarButtonItemStylePlain target:nil action:nil];
    self.navigationItem.title = @"title";
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)bu:(id)sender {
    UIViewController *vc = [ViewController2 new];
    [self addChildViewController:vc];
    [self.view addSubview:vc.view];
//    [vc didMoveToParentViewController:self];
}

@end

File: TERespinseChains/TERespinseChains/ViewController.swift
//
//  ViewController.swift
//  TERespinseChains
//
//  Created by bo on 01/01/2017.
//  Copyright © 2017 bo. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    let abc : String
    
    
    required override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        self.abc = "as"
        print("\(self.abc)")
        print("\(abc)")
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        self.abc = "as"
//        print("\(self)")
        print("\(abc)")
        super.init(coder: aDecoder)
        
    }
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

